### PR TITLE
Added support for tags

### LINKED ISSUE #148: Support for tags
### 🎯 Goals

- The goal is to allow checking out a repository to a tag (version) instead of a branch.
- The tag can be unknown, so we want to check out the latest created.

### 📗 Story

- **Given**: I am a developer who uses the https://github.com/ckeditor/ckeditor5-dev repository in a project.
	```json
	{
	  "packages": "external/",
	  "dependencies": {
	    "ckeditor5-dev": "ckeditor/ckeditor5-dev@latest"
	  }
	}
	```
- **When**: I execute `mrgit sync`.
- **Then**: The tool checks out the `ckeditor5-dev` repository on the `v31.1.11` tag.

### 💡 Hints

- When creating the issue, the latest version in `ckeditor5-dev` is `v31.1.11`
	image
- We should also support defining a particular tag: `"ckeditor5-dev": "ckeditor/ckeditor5-dev@v31.0.0"`.
- The `@latest` has a specific meaning - find the latest tag name and use it.

### PR DESCRIPTION
**Pull request by przemyslaw-zan**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Added support for tags. Closes #148.

---

### Additional information

*For example – encountered issues, assumptions you had to make, other affected tickets, etc.*

### CODE REVIEW

**Comment by pomek** [comment]
A few notes without scanning the code:

* It would be good to exclude repositories checked-out tags from `push`, `pull`, `commit`, and `close` commands.
	* `push`:
		image
	* `pull`:
		image
* What the `L` symbol stands for?
	image
	* I would go with `@` or `T` (tags). L (latest) does not mean anything in the entire context of the status command.
* `mrgit save` produces invalid results now.
	```json
	"ckeditor5-dev": "ckeditor/ckeditor5-dev@latest#dccbeb5"
	```
	* `mrgit save --branch` fails even for repositories with no tags.
	* Everything works fine on master.

**Comment by pomek** [comment]
> `mrgit save --branch` fails even for repositories with no tags.

It creates the following entry: `"ckeditor5-dev": "ckeditor/ckeditor5-dev#HEAD (no branch)"` when checked-out tag.

But it can be extracted to a follow-up issue.

File: lib/commands/status.js [hunk 1]
```diff
@@ -42,22 +42,29 @@ module.exports = {
 
 		const promises = [
 			execCommand.execute( getExecData( 'git rev-parse HEAD' ) ),
-			execCommand.execute( getExecData( 'git status --branch --porcelain' ) )
+			execCommand.execute( getExecData( 'git status --branch --porcelain' ) ),
+			execCommand.execute( getExecData( 'git describe --abbrev=0 --tags' ) ),
+			execCommand.execute( getExecData( 'git tag --sort=committerdate' ) )
```
Comment (pomek) [inline lib/commands/status.js:47]:
```
"mrgit": "https://github.com/cksource/mrgit.git@latest"
```

image

This is crazy. We need to find a new way to extract the latest tag.

Comment (pomek) [reply lib/commands/status.js]: Addressed in 0b9fc08.

**Review by pomek — APPROVED** [review]
LGTM.

File: lib/commands/sync.js [hunk 2]
```diff
@@ -73,8 +74,21 @@ module.exports = {
 			.then( response => {
 				log.concat( response.logs );
 			} )
-			.then( () => {
-				return execCommand.execute( getExecData( `git checkout ${ data.repository.branch }` ) );
+			.then( async () => {
+				let checkoutValue;
+
+				if ( !data.repository.tag ) {
```
[the remark this answers is not in the document]
Comment (przemyslaw-zan) [reply lib/commands/sync.js:80]: They are both just a bit different, making a function out of the two would require bunch of arguments and would make more mess than anything, not worth it IMO

**Pull request merged**